### PR TITLE
Issue #7726: Update AbstractChecks to log DetailAST - AvoidStaticImport

### PR DESCRIPTION
**Pull request by gaurabdg**
Fixes Issue #7726 
Log Update:
https://github.com/checkstyle/checkstyle/blob/5ec8cfe4ded197c67f0f4bb8edfddde1d34d78c1/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheck.java#L128

Diff Report: https://gaurabdg.github.io/checkstyle-tester-reports/regression/AvoidStaticImport/diff/index.html

### CODE REVIEW

**Comment by gaurabdg** [comment]
@rnveach Please review.

File: src/it/resources/org/checkstyle/suppressionxpathfilter/avoidstaticimport/SuppressionXpathRegressionAvoidStaticImport1.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.checkstyle.suppressionxpathfilter.avoidstaticimport;
+
+import static javax.swing.WindowConstants.*; // warn
+
+public class SuppressionXpathRegressionAvoidStaticImport1 {
+    public static void main(String[] args) {
```
Comment (strkkk) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/avoidstaticimport/SuppressionXpathRegressionAvoidStaticImport1.java]: No need for method, can be removed

File: src/it/resources/org/checkstyle/suppressionxpathfilter/avoidstaticimport/SuppressionXpathRegressionAvoidStaticImport2.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.checkstyle.suppressionxpathfilter.avoidstaticimport;
+
+import static java.io.File.createTempFile; // warn
+
+public class SuppressionXpathRegressionAvoidStaticImport2 {
+    public static void main(String[] args) {
```
Comment (strkkk) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/avoidstaticimport/SuppressionXpathRegressionAvoidStaticImport2.java]: same here

File: src/it/resources/org/checkstyle/suppressionxpathfilter/avoidstaticimport/SuppressionXpathRegressionAvoidStaticImport1.java (hunk 1 above)
Comment (gaurabdg) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/avoidstaticimport/SuppressionXpathRegressionAvoidStaticImport1.java]: done

File: src/it/resources/org/checkstyle/suppressionxpathfilter/avoidstaticimport/SuppressionXpathRegressionAvoidStaticImport2.java (hunk 2 above)
Comment (gaurabdg) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/avoidstaticimport/SuppressionXpathRegressionAvoidStaticImport2.java]: done

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java [hunk 3]
```diff
@@ -50,17 +50,17 @@ public void testDefaultOperation()
...
-            "25: " + getCheckMessage(MSG_KEY, "javax.swing.WindowConstants.*"),
-            "26: " + getCheckMessage(MSG_KEY, "javax.swing.WindowConstants.*"),
-            "27: " + getCheckMessage(MSG_KEY, "java.io.File.createTempFile"),
-            "28: " + getCheckMessage(MSG_KEY, "java.io.File.pathSeparator"),
-            "29: " + getCheckMessage(MSG_KEY, "java.lang.Math.E"),
-            "30: " + getCheckMessage(MSG_KEY, "java.lang.Math.sqrt"),
-            "31: " + getCheckMessage(MSG_KEY,
+            "23:27: " + getCheckMessage(MSG_KEY, "java.io.File.listRoots"),
+            "25:42: " + getCheckMessage(MSG_KEY, "javax.swing.WindowConstants.*"),
+            "26:42: " + getCheckMessage(MSG_KEY, "javax.swing.WindowConstants.*"),
```
Comment (strkkk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java]: This is outside the scope of the issue, but please - this import in file and duplicated violation, please remove them.

Comment (gaurabdg) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java [hunk 4]
```diff
@@ -50,17 +50,16 @@ public void testDefaultOperation()
...
+            "25:42: " + getCheckMessage(MSG_KEY, "javax.swing.WindowConstants.*"),
+            "26:27: " + getCheckMessage(MSG_KEY, "java.io.File.createTempFile"),
+            "27:27: " + getCheckMessage(MSG_KEY, "java.io.File.pathSeparator"),
+            "28:29: " + getCheckMessage(MSG_KEY, "java.lang.Math.E"),
+            "29:29: " + getCheckMessage(MSG_KEY, "java.lang.Math.sqrt"),
+            "30:113: " + getCheckMessage(MSG_KEY,
                 "com.puppycrawl.tools.checkstyle.checks.imports."
                     + "avoidstaticimport.InputAvoidStaticImportNestedClass.InnerClass"),
-            "32: " + getCheckMessage(MSG_KEY,
+            "31:124: " + getCheckMessage(MSG_KEY,
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java]: Please confirm why line number changed from 32 to 31. 
https://github.com/checkstyle/checkstyle/blob/555199e6bbb338ef84afffd1a288100041f0aae9/src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/avoidstaticimport/InputAvoidStaticImportDefault.java#L31
I am not seeing a reason why in the file as it doesn't span more than 1 line.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/avoidstaticimport/InputAvoidStaticImportDefault.java [hunk 5]
```diff
@@ -22,7 +22,6 @@
 import javax.swing.BorderFactory;
 import static java.io.File.listRoots;
 
-import static javax.swing.WindowConstants.*;
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/avoidstaticimport/InputAvoidStaticImportDefault.java]: Why is this removed?

Comment (rnveach) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/avoidstaticimport/InputAvoidStaticImportDefault.java]: @strkkk We are not expecting inputs to be changed in these issues.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java (hunk 4 above)
Comment (gaurabdg) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java]: I was asked by @strkkk to remove the input duplicate import

Comment (gaurabdg) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java]: Shall I revert?

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/avoidstaticimport/InputAvoidStaticImportDefault.java (hunk 5 above)
Comment (strkkk) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/avoidstaticimport/InputAvoidStaticImportDefault.java]: @rnveach it is a duplicated import.
Line 26 in same file contains exactly same import, so this line is redundant.
It is out of scope of this issue, but I asked @gaurabdg to remove it,

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java (hunk 4 above)
Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java]: https://github.com/checkstyle/checkstyle/pull/476
I don't see any reason for the duplicate import. Normally we wouldn't change inputs like this as they may be testing something. Additionally, to remove a line from a file with violations will cause the entire test's line numbers to change.

If you want to keep this change, it must be another commit since it is outside the scope of this issue.

Comment (gaurabdg) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java]: Okay, so it should be a different PR altogether or 2 commits in this PR?

Comment (strkkk) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java]: second commit with message like "minor: remove duplicated violation from AvoidStaticImport"

Comment (gaurabdg) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStaticImportCheckTest.java]: Sure
